### PR TITLE
crimson/osd/pg: rebuild missing set when a new interval is created

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Fixes: https://tracker.ceph.com/issues/58339
Signed-off-by: Xuehan Xu <xxhdx1985126@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by xxhdx1985126** [comment]
This pr shares the commit 9bf30b505ff603ae56b7a84db84c0ce90bb0f5c7 with #49020

**Comment by xxhdx1985126** [comment]
jenkins test make check

File: src/crimson/osd/osd_operations/peering_event.cc [hunk 1]
```diff
@@ -89,9 +89,12 @@ seastar::future<> PeeringEvent<T>::with_pg(
       return this->template enter_stage<interruptor>(
 	BackfillRecovery::bp(*pg).process);
     }).then_interruptible([this, pg, &shard_services] {
-      pg->do_peering_event(evt, ctx);
-      that()->get_handle().exit();
-      return complete_rctx(shard_services, pg);
+      return interruptor::async([this, pg] {
+	pg->do_peering_event(evt, ctx);
```
Comment (cyx1231st) [inline src/crimson/osd/osd_operations/peering_event.cc:93]: Are you able to elaborate in comments that where is the dependencies that call `seastar::future::wait()` in the `do_peering_event()` and `handle_advance_map()` implemetations?

I think it's error-prone and thus important to be explicit that certain methods must be wrapped by `async()` when want to call it.

Comment (xxhdx1985126) [reply src/crimson/osd/osd_operations/peering_event.cc:93]: Yeah, will do:-)

Comment (athanatos) [reply src/crimson/osd/osd_operations/peering_event.cc:93]: @xxhdx1985126 If I'm interpreting this correctly, the idea is to all PeeringState callbacks to freely run IO.  Reasonable, but I want you to invoke ::async right at the call into the PeeringState, not in the events.  do_peering_event and handle_advance_map should be doing the wrapping as close as possible to the PeeringState state machine invocation, and you need to add comments explaining very obviously why.

Comment (athanatos) [reply src/crimson/osd/osd_operations/peering_event.cc:93]: Indeed, if you can find a single place to do the wrapping, that would be ideal.

Comment (xxhdx1985126) [reply src/crimson/osd/osd_operations/peering_event.cc:93]: Yes, will do

Comment (xxhdx1985126) [reply src/crimson/osd/osd_operations/peering_event.cc]: done, please take a look again

**Comment by xxhdx1985126** [comment]
jenkins test api

**Comment by xxhdx1985126** [comment]
jenkins test api

**Review by athanatos — APPROVED** [review]

**Pull request merged**